routes/dao_coin_exchange_with_fees.go

```diff
@@ -0,0 +1,1602 @@
+				uint256LimitReceiveAmount := uint256.NewInt()
+				if overflow := uint256LimitReceiveAmount.SetFromBig(bigLimitReceiveAmount); overflow {
+					return nil, fmt.Errorf("HandleMarketOrder: Overflow calculating limit receive amount")
+				}
+				limitReceiveAmount, err = CalculateStringDecimalAmountFromBaseUnitsSimple(
+					req.BaseCurrencyPublicKeyBase58Check, uint256LimitReceiveAmount)
+				if err != nil {
+					return nil, fmt.Errorf("HandleMarketOrder: Problem calculating limit receive amount: %v", err)
+				}
+			}
+
+			// Set all the values we calculated
+			res.LimitAmount = limitAmount
+			res.LimitAmountCurrencyType = CurrencyTypeQuote
+			res.LimitAmountInUsd = convertToUsd(limitAmount)
+			res.LimitReceiveAmount = limitReceiveAmount
+			res.LimitReceiveAmountCurrencyType = CurrencyTypeBase
+			res.LimitReceiveAmountInUsd = "" // dont convert base currency to usd
+			res.LimitPriceInQuoteCurrency = priceStrQuote
+			res.LimitPriceInUsd = convertToUsd(priceStrQuote)
+		}
+
+		return res, nil
+	} else {
+		// We already have the txn that executes the order from previously
+		// Connect it to our UtxoView for validation
+		bb, err := hex.DecodeString(orderRes.TransactionHex)
+		if err != nil {
+			return nil, fmt.Errorf("HandleMarketOrder: Problem decoding txn hex: %v", err)
+		}
+		orderTxn := &lib.MsgDeSoTxn{}
+		if err := orderTxn.FromBytes(bb); err != nil {
+			return nil, fmt.Errorf("HandleMarketOrder: Problem parsing txn: %v", err)
+		}
+		_, _, _, _, err = utxoView.ConnectTransaction(
+			orderTxn, orderTxn.Hash(), fes.blockchain.BlockTip().Height,
+			fes.blockchain.BlockTip().Header.TstampNanoSecs,
+			false, false)
+		if err != nil {
+			return nil, fmt.Errorf("HandleMarketOrder: Problem connecting transaction: %v", err)
+		}
+
+		// Now we need to deduct the fees from the user's balance.
+		// For each trading fee we need to pay, construct a transfer txn that sends the amount
+		// from the transactor directly to the person receiving the fee.
+		transferTxns := []*lib.MsgDeSoTxn{}
+		for pkid, feeBaseUnits := range feeBaseUnitsByPubkey {
+			receiverPubkeyBytes, _, err := lib.Base58CheckDecode(pkid)
+			if err != nil || len(receiverPubkeyBytes) != btcec.PubKeyBytesLenCompressed {
+				return nil, fmt.Errorf("HandleMarketOrder: Problem decoding public key %s: %v",
+					pkid, err)
+			}
+			// Try and create the TransferDaoCoin transaction for the user.
+			//
+			// TODO: Add ExtraData to the transaction to make it easier to report it as an
+			// earning to the user who's receiving the fee.
+			txn, _, _, _, err := fes.blockchain.CreateDAOCoinTransferTxn(
+				transactorPubkeyBytes,
+				&lib.DAOCoinTransferMetadata{
+					ProfilePublicKey:       quoteCurrencyPkidBytes,
+					ReceiverPublicKey:      receiverPubkeyBytes,
+					DAOCoinToTransferNanos: *feeBaseUnits,
+				},
+				// Standard transaction fields
+				req.MinFeeRateNanosPerKB, fes.backendServer.GetMempool(), nil)
```

Comment:
once again this breaks this node-level transaction fee stuff we worked on.

Reply:
Fixed.